src/components/tables/RowActions/Shared/Confirmation.tsx

```diff
@@ -1,29 +1,51 @@
 import { List, ListItem, Typography } from '@mui/material';
 import { ReactNode } from 'react';
+import { SelectTableStoreNames } from 'stores/names';
+import NestedListItem, { SettingMetadata } from './NestedListItem';
 
-interface RowActionConfirmationprops {
+interface RowActionConfirmationProps {
     message: ReactNode;
     selected: any; //SelectableTableStore['selected'];
+    selectableTableStoreName?:
+        | SelectTableStoreNames.CAPTURE
+        | SelectTableStoreNames.COLLECTION
+        | SelectTableStoreNames.COLLECTION_SELECTOR
+        | SelectTableStoreNames.MATERIALIZATION;
+    settings?: SettingMetadata[];
 }
 
 function RowActionConfirmation({
     message,
     selected,
-}: RowActionConfirmationprops) {
+    selectableTableStoreName,
+    settings,
+}: RowActionConfirmationProps) {
+    const renderListItems =
+        settings && settings.length > 0 && selectableTableStoreName
+            ? (value: string) => (
+                  <NestedListItem
+                      key={`confirmation-selected-items-${value}`}
+                      catalogName={value}
+                      selectableTableStoreName={selectableTableStoreName}
+                      settings={settings}
+                  />
+              )
+            : (value: string) => (
+                  <ListItem
+                      component="div"
+                      key={`confirmation-selected-items-${value}`}
+                  >
+                      <Typography component="span" sx={{ fontWeight: 500 }}>
+                          {value}
+                      </Typography>
+                  </ListItem>
+              );
```

Comment:
Given how different these are... I wonder if we should just not even try to make them run through the same code? Would it be better to just pass the list renderer component as a prop?